lib/DeviceManagerPlugin.ts

```diff
@@ -201,6 +204,40 @@ export class DeviceManagerPlugin extends Plugin {
       "device-manager/device:beforeSearch": this.pipeCheckEngine.bind(this),
       "device-manager/device:beforeUpdate": this.pipeCheckEngine.bind(this),
       "generic:document:beforeWrite": [],
+      "generic:document:afterGet": async (documents, request: Request) => {
+        if (
+          (request.input.args.collection === "assets" ||
+            request.input.args.collection === "devices") &&
+          request.input.args.options?.prettify
+        ) {
+          for (const document of documents) {
+            if (document._source?.metadata) {
+              document._source.metadata =
+                this.assetCategoryService.formatMetadataForGet(
+                  document._source.metadata
+                );
+            }
+          }
+        }
+        return documents;
+      },
+      "generic:document:beforeUpdate": async (documents, request: Request) => {
+        if (
+          (request.input.args.collection === "assets" ||
+            request.input.args.collection === "devices") &&
+          request.input.args.options?.prettify
+        ) {
+          for (const document of documents) {
+            if (document._source?.metadata) {
+              document._source.metadata =
+                this.assetCategoryService.formatMetadataForES(
+                  document._source.metadata
+                );
+            }
+          }
+        }
+        return documents;
```

Comment:
This part could be DRY'ied

Reply:
I can do something like 

```
pipeFunction(myFunction)
        if (
          (request.input.args.collection === "assets" ||
            request.input.args.collection === "devices") &&
          request.input.args.options?.prettify
        ) {
          for (const document of documents) {
            if (document._source?.metadata) {
              document._source.metadata =
                myFunction(
                  document._source.metadata
                );
            }
          }
        }
...
this.pipes = {
     ....
      "generic:document:afterGet": async (documents, request: Request) => {
       return pipeFunction( this.assetCategoryService.formatMetadataForGet);}
      "generic:document:beforeUpdate": async (documents, request: Request) => {
       return pipeFunction(this.assetCategoryService.formatMetadataForES);}
};
```
It's more concise, but, in my opinion, (even with better naming) less readable.